backend/jack_backend.cpp

```diff
@@ -176,6 +184,17 @@ void JackBackend::setVolume( QString channel, QString output, float volume ) {
 		volumes[ channel ][ output ] = volume;
 }
 
+void JackBackend::setVolumeNew(QString channel, QString output, float volume) {
```

Comment:
Why are you adding a new function? As you are replacing every call to `setVolume` with `setVolumeNew` anyway, you could and should just change the implementation of `setVolume`.

You would get one initial ramp from 0-level to the initial values upon start if you initialize the 'current' volumes at zero…

Reply:
[I use setVolume in the process callback](https://github.com/dsheeler/jackmix-1/blob/3456e8ffc39f6b2d1c54f75909346657c5483e48/backend/jack_backend.cpp#L369-L371).  setVolumeNew is used when some change in volume is requested, and setVolume is called in the process callback when the request is satisfied and the volume is set to the new volume.  Does that make sense?  I think something is necessary to distinguish the two operations.  Maybe `setVolumeNew` would be better named `requestVolume`.  Also, I could add a flag to `setVolume` to make the distinction instead of adding `setVolumeNew`.  I could be confused here, of course :).

Reply:
If setVolume is now only used in the process callback, it should be a private function and can be renamed to something like `setFinalVolume`. Then `setVolume` can be used for the external interface. And it would keep the API compatible.

Reply:
OK, I'm back at the coal face!

I agree with @kampfschlaefer about the private function being used to avoid API changes. I'll take a closer look and get back to you if I change my mind.

Reply:
I agree, too :).